src/Builder/SplitItemBuilder.php

```diff
@@ -139,4 +168,42 @@ public function build() : SplitItem
     {
         return $this->splitItem;
     }
+
+    public function getCheckboxStyle() : CheckboxStyle
+    {
+        return $this->checkboxStyle;
+    }
+
+    public function setCheckboxStyle(CheckboxStyle $style) : self
+    {
+        $this->checkboxStyle = $style;
+
+        return $this;
+    }
+
+    public function modifyCheckboxStyle(callable $itemCallable) : self
```

Comment:
I actually think we should drop this. It's took me a while but I think I get what you are doing. If you change the styles in a splititem builder, they persist back to the parent menu. I think it makes the code very confusing. I would rather we just not allow to modify the styles in the `SplitItem`. Or rather if you modify the styles, it only applied to the split item and not the parent menu.

Reply:
Style is always limited only to the current scope and any children:

```
<?php

use PhpSchool\CliMenu\Builder\SplitItemBuilder;
use PhpSchool\CliMenu\CliMenu;
use PhpSchool\CliMenu\Builder\CliMenuBuilder;
use PhpSchool\CliMenu\Style\CheckboxStyle;

require_once(__DIR__ . '/../vendor/autoload.php');

$itemCallable = function (CliMenu $menu) {
    echo $menu->getSelectedItem()->getText();
};

$menu = (new CliMenuBuilder)
    ->setTitle('Select a Language')
    ->addCheckboxItem('Rust', $itemCallable)
    ->addCheckboxItem('C++', $itemCallable)
    ->addCheckboxItem('Go', $itemCallable)
    ->addCheckboxItem('Java', $itemCallable)
    ->addCheckboxItem('C', $itemCallable)
    ->addSplitItem(function (SplitItemBuilder $b) use ($itemCallable) {
        $b->setGutter(5)
            ->modifyCheckboxStyle(function (CheckboxStyle $style) {
                $style->setMarkerOff('ooo')
                    ->setMarkerOn('xxx');
            })
            ->addCheckboxItem('Rust', $itemCallable)
            ->addCheckboxItem('C++', $itemCallable)
            ->addCheckboxItem('Go', $itemCallable)
            ->addCheckboxItem('Java', $itemCallable)
            ->addCheckboxItem('C', $itemCallable)
        ;
    })
    ->build();

$menu->open();
```

The parent should never be affected from the `modify*Style()` methods:

![Peek 2019-12-19 16-26](https://user-images.githubusercontent.com/589398/71215037-478e0400-227c-11ea-94ea-bbdd04eff3aa.gif)

Reply:
Actually you're right, there's a bug where changing in a child updates the parent:

```
<?php

use PhpSchool\CliMenu\Builder\SplitItemBuilder;
use PhpSchool\CliMenu\CliMenu;
use PhpSchool\CliMenu\Builder\CliMenuBuilder;
use PhpSchool\CliMenu\Style\CheckboxStyle;

require_once(__DIR__ . '/../vendor/autoload.php');

$itemCallable = function (CliMenu $menu) {
    echo $menu->getSelectedItem()->getText();
};

$menu = (new CliMenuBuilder)
    ->setTitle('Select a Language')
    ->addCheckboxItem('1-A', $itemCallable)
    ->addCheckboxItem('1-B', $itemCallable)
    ->addSplitItem(function (SplitItemBuilder $b) use ($itemCallable) {
        $b->setGutter(5)
            ->addCheckboxItem('2-A', $itemCallable)
            ->modifyCheckboxStyle(function (CheckboxStyle $style) {
                $style->setMarkerOff('o ')
                    ->setMarkerOn('x ');
            })
            ->addCheckboxItem('2-B', $itemCallable)
            ->addSubMenu('Options', function (CliMenuBuilder $b) use ($itemCallable) {
                $b->setTitle('CLI Menu > Options')
                    ->addCheckboxItem('Rust', $itemCallable)
                    ->addCheckboxItem('C++', $itemCallable)
                    ->addCheckboxItem('Go', $itemCallable)
                    ->addCheckboxItem('Java', $itemCallable)
                    ->addCheckboxItem('C', $itemCallable)
                    ->addLineBreak('-');
            })
            ->addCheckboxItem('2-C', $itemCallable)
            ->addCheckboxItem('2-D', $itemCallable)
            ->addCheckboxItem('2-E', $itemCallable)
        ;
    })
    ->addCheckboxItem('1-C', $itemCallable)
    ->addCheckboxItem('1-D', $itemCallable)
    ->addCheckboxItem('1-E', $itemCallable)
    ->build();

$menu->open();
```

I'll fix and add a test.

Reply:
Actually, I think the problem is that I had a fundamental misunderstanding on the SplitItem.

I was thinking this was a separate menu ala submenu, but it's simply a horizontal partitioner of the current menu.

You're completely right, SplitItemBuilder shouldn't have any of the radio/checkbox style methods, and should not have its own style instance(s).